src/Generating/FinishGen.cpp

```diff
@@ -1395,3 +1395,369 @@ eMonsterType cFinishGenPassiveMobs::GetRandomMob(cChunkDesc & a_ChunkDesc)
+	double stepZ = (z2 - z1) / static_cast<double>(a_NestSize);
+	double stepR = M_PI / static_cast<double>(a_NestSize);
+	double size = static_cast<double>(a_NestSize) / 16.0;
+	for (int i = 0; i < a_NestSize; ++i)
+	{
+		double iDbl = static_cast<double>(i);
+		double sphereX = x1 + stepX * iDbl;
+		double sphereY = y1 + stepY * iDbl;
+		double sphereZ = z1 + stepZ * iDbl;
+		double radius = (sin(stepR * iDbl) + 1.0) * size + 1.0;
+		imprintSphere(a_ChunkDesc, sphereX, sphereY, sphereZ, radius, a_OreType, a_OreMeta);
+	}  // for i
+}
+
+
+
+
+
+void cFinishGenOrePockets::imprintSphere(
+	cChunkDesc & a_ChunkDesc,
+	double a_SphereX, double a_SphereY, double a_SphereZ, double a_Radius,
+	BLOCKTYPE a_OreType, NIBBLETYPE a_OreMeta
+)
+{
+	// Get the sphere's bounding box, unioned with the chunk's bounding box (possibly empty):
+	int baseX = a_ChunkDesc.GetChunkX() * cChunkDef::Width;
+	int baseZ = a_ChunkDesc.GetChunkZ() * cChunkDef::Width;
+	int minX = std::max(FloorC(a_SphereX - a_Radius), baseX);
+	int minY = std::max(FloorC(a_SphereY - a_Radius), 0);
+	int minZ = std::max(FloorC(a_SphereZ - a_Radius), baseZ);
+	int maxX = std::min(CeilC(a_SphereX + a_Radius), baseX + cChunkDef::Width - 1);
+	int maxY = std::min(CeilC(a_SphereY + a_Radius), cChunkDef::Height - 1);
+	int maxZ = std::min(CeilC(a_SphereZ + a_Radius), baseZ + cChunkDef::Width - 1);
+
+	/*
+	// DEBUG:
+	int blockX = FloorC(a_SphereX);
+	int blockY = FloorC(a_SphereY);
+	int blockZ = FloorC(a_SphereZ);
+	if (
+		(blockX >= baseX) && (blockX < baseX + cChunkDef::Width) &&
+		(blockY >= 0) && (blockY < cChunkDef::Height) &&
+		(blockZ >= baseZ) && (blockZ < baseZ + cChunkDef::Width)
+	)
+	{
+		// LOGD("Imprinting a sphere center at {%d, %d, %d}", blockX, blockY, blockZ);
+		a_ChunkDesc.SetBlockTypeMeta(blockX - baseX, blockY, blockZ - baseZ, a_OreType, a_OreMeta);
+	}
+	return;
+	//*/
+
+	// Imprint the parts of the sphere intersecting the chunk:
+	double radiusSq = a_Radius * a_Radius / 4.0;
+	for (int y = minY; y <= maxY; y++)
+	{
+		double relY = static_cast<double>(y) + 0.5 - a_SphereY;
+		double relYSq = relY * relY;
+		if (relYSq > radiusSq)
+		{
+			// outside the sphere, bail out
+			continue;
+		}
+		for (int z = minZ; z <= maxZ; z++)
+		{
+			double relZ = static_cast<double>(z) + 0.5 - a_SphereZ;
+			double relZSq = relZ * relZ;
+			if (relZSq + relYSq > radiusSq)
+			{
+				// outside the sphere, bail out
+				continue;
+			}
+			for (int x = minX; x <= maxX; x++)
+			{
+				double relX = static_cast<double>(x) + 0.5 - a_SphereX;
+				double relXSq = relX * relX;
+				if (relZSq + relYSq + relXSq > radiusSq)
+				{
+					// outside the sphere, bail out
+					continue;
+				}
+				int bX = x - baseX;
+				int bZ = z - baseZ;
+				auto blockType = a_ChunkDesc.GetBlockType(bX, y, bZ);
+				if ((blockType == E_BLOCK_STONE) || (blockType == E_BLOCK_NETHERRACK))
+				{
+					a_ChunkDesc.SetBlockTypeMeta(bX, y, bZ, a_OreType, a_OreMeta);
```

Comment:
Only if it can access the code. If you're not doing LTO, because the state of LTO on *nix is abysmal, then any function not in the same translation unit, can't be inlined. cChunkDesc::GelBlockType is in a different translation unit from FinishGen.cpp